packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -458,41 +460,48 @@ class IrisGridTableModel extends IrisGridModel {
     return '';
   }
 
-  getColumnsWithLayoutHints = memoize((columns, hints) => {
-    if (hints) {
+  getColumnsWithLayoutHints = memoize((columns, hints, uiFrozenColumns) => {
+    if (hints || uiFrozenColumns.length) {
       const columnMap = new Map();
       columns.forEach(col => columnMap.set(col.name, col));
 
       let frontColumns = [];
       let backColumns = [];
       let frozenColumns = [];
 
-      if (hints.frontColumns) {
+      if (hints?.frontColumns) {
         frontColumns = hints.frontColumns
           .map(name => columnMap.get(name))
           .filter(Boolean);
       }
-      if (hints.backColumns) {
+      if (hints?.backColumns) {
         backColumns = hints.backColumns
           .map(name => columnMap.get(name))
           .filter(Boolean);
       }
-      if (hints.frozenColumns) {
+      if (hints?.frozenColumns) {
         frozenColumns = hints.frozenColumns
           .map(name => columnMap.get(name))
           .filter(Boolean);
       }
 
       if (
-        frontColumns.length !== (hints.frontColumns?.length ?? 0) ||
-        backColumns.length !== (hints.backColumns?.length ?? 0) ||
-        frozenColumns.length !== (hints.frozenColumns?.length ?? 0)
+        frontColumns.length !== (hints?.frontColumns?.length ?? 0) ||
+        backColumns.length !== (hints?.backColumns?.length ?? 0) ||
+        frozenColumns.length !== (hints?.frozenColumns?.length ?? 0)
       ) {
         throw new Error(
           'Layout hints are invalid (contain invalid column names)'
         );
       }
 
+      // Handle frozen columns by user UI action.
+      if (uiFrozenColumns.length) {
+        frozenColumns = uiFrozenColumns
```

Comment:
We need to be able to freeze in addition to any existing hints, as well as unfreeze any columns from hints. Not sure how we do it with this data construct?